packages/runtime/datastore/src/remoteChannelContext.ts

```diff
@@ -100,16 +106,19 @@ export class RemoteChannelContext implements IChannelContext {
 				this.id,
 			);
 
-			// Send all pending messages to the channel
-			assert(this.pending !== undefined, 0x23f /* "pending undefined" */);
-			for (const message of this.pending) {
-				this.services.deltaConnection.process(message, false, undefined /* localOpMetadata */);
+			assert(this.pendingMessagesState !== undefined, "pending messages queue is undefined");
```

Comment:
other than slightly better error message, asserts like that do not have much value (but add to bundle size) - the next like would crash either way. That said, JS tooling sucks, and bucketing provided by assert tags has some value...